### PR TITLE
Enable GZIP

### PR DESCRIPTION
**Pull request by mbolt35**
Enable GZIP Encoding for specific mimeType responses over 50kb.

### CODE REVIEW

**Comment by mbolt35** [comment]
NGINX Docs here: https://docs.nginx.com/nginx/admin-guide/web-server/compression/

**Review by nikovacevic — APPROVED** [review]
LGTM

**Comment by mbolt35** [comment]
@AjayTripathy Before I merge this, wanted to see if you would take a look. Just want to make sure this doesn't break SAML.

**Pull request merged**
